tests/integration_test/core_mqtt_system_test.c

```diff
@@ -781,6 +898,26 @@ static MQTTStatus_t publishToTopic( MQTTContext_t * pContext,
                          packetId );
 }
 
+static void clearRetainedMessage( MQTTContext_t * pContext,
+                                  const char * pTopic )
+{
+    assert( pContext != NULL );
```

Comment:
Can also assert pTopic != NULL?

Reply:
Sure, but since the other helpers i hadn't here as well.

Reply:
I would like to understand the reasoning for checking the one parameter here but not the other. Saying this was done incorrectly elsewhere is not sufficient reason to dismiss this question, if it is wrong elsewhere that also needs to be fixed.

That said it seems strange to check parameters on static functions, it is often better to ensure we always pass valid date into these and assert earlier (before calling the function something else must have failed - shift left here)

Reply:
This is fixed in https://github.com/aws/amazon-freertos/pull/2755/commits/e349b75ed0a722ed8a367695aa0550e040f2312a

Reply:
This is simply a defensive developer check on parameters that should never be NULL.